activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+*   Exclude ids of included records with #ids
```

Comment:
```suggestion
*   Return only unique ids from ActiveRecord::Calculations#ids
```
We should clarify that we're excluding duplicated ids here (in other words, returning only unique ids)

Reply:
I think my tests may have been a bit misleading. To reiterate:

> This Pull Request changes ActiveRecord::Calculations#ids to only pluck the ids of the base model, ignoring eager loaded, preloaded and included associations.

I'm recommending that we ignore preloaded record ids altogether. So it's not just excluding duplicates, it's **only** plucking the base model's ids. This may not be apparent from the tests since the associations created using `5.times` would probably contain ids matching existing base model records.

Reply:
But the point of not applying the join dependency is so that we don't get back the same id (for the base relation) more than once, right? Before these changes (using your tests as an example), we'd end up with a relation containing 5 Active Records for Company with id 1, with each record tied to a Contract. We're not excluding the ids on the `Contract` records, we're excluding the duplicated ids of the base relation (Company).

Reply:
Ah yes of course I see what you mean now, that is correct. I think having been away from this for a while I managed to forget what the actual issue was haha. I have actioned your suggestion!